### PR TITLE
Change Sandbox.SetKey() in case of shared osl.Sandbox

### PR DESCRIPTION
**Pull request by aboch**
- In case of `useDefaultSandbox`, if SetKey finds
  an existing osl.Sandbox it is not an error
  condition and no more processing should be done

Signed-off-by: Alessandro Boch <aboch@docker.com>

### CODE REVIEW

**Comment by mavenugo** [comment]
@aboch with this change, though it is not failing to create container in host mode, there is no connectivity. infact the namespace is missing all the host interfaces.  this fix needs a rework.
Also, please test the changes integrating with docker.
ping @mrjana

**Comment by estesp** [comment]
As @mavenugo said, there is a continuing problem with host networking not having the network interfaces/connectivity.  Container mode is actually working with this change, but not `--net=host`.  Actually, if it helps, in `--net=host` mode the `/proc/self/ns/net` is most definitely not the same as the host, but in `--net=container:<somecontainer>` it does actually match the net namespace of the other container.

**Comment by mavenugo** [comment]
@estesp thats correct. `--net=container` mode is handled differently and should work without any extra changes. But I forgot to handle the `--net=host` case in my original change (as it was originally intended for userns support). In any case, this PR needs some rework to handle this scenario.

**Comment by aboch** [comment]
@estesp @mavenugo I updated the new diffs. I will test them within docker later, no time now.

File: sandbox.go [hunk 1]
```diff
@@ -279,25 +279,28 @@ func (sb *sandbox) SetKey(basePath string) error {
...
-	if sb.osSbox != nil {
-		sb.Unlock()
-		return types.ForbiddenErrorf("failed to set sandbox key : already assigned")
-	}
+	osSbox := sb.osSbox
 	sb.Unlock()
-	osSbox, err := osl.GetSandboxForExternalKey(basePath, sb.Key())
-	if err != nil {
-		return err
+	if osSbox != nil && sb.config.useDefaultSandBox {
```
Comment (mavenugo) [inline sandbox.go]: i believe you meant `!sb.config.useDefaultSandBox`

**Comment by mavenugo** [comment]
@aboch in addition to the above comment, it is still not working. :(

**Comment by mavenugo** [comment]
@aboch @estesp PTAL https://github.com/docker/docker/pull/16305#issuecomment-140618455. I dont think we need this patch any more. We can close it if you agree.

**Comment by aboch** [comment]
Thanks @mavenugo Closing this

**Pull request closed without merging**